llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -7657,6 +7717,13 @@ static bool isGuaranteedNotToBeUndefOrPoison(
       }
       if (IsWellDefined)
         return true;
+
+      bool StartNeedsFreeze;
+      if (canFoldFreezeIntoRecurrence(const_cast<PHINode *>(PN),
```

Comment:
Is it worth having a limit on the number of incoming values to make sure we keep compile time under control?

Reply:
I ran ctmark and the difference is negligible, but I suppose it's not a panacea, so I can if we think it's necessary. Although I'm not sure what a reasonable number is? tbh the thing I care about it's only 2 incoming values so I could restrict it to that.